app/components/upload-app/status/details/index.ts

```diff
@@ -0,0 +1,86 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import ENUMS from 'irene/enums';
+import dayjs from 'dayjs';
+import { action } from '@ember/object';
+import SubmissionModel from 'irene/models/submission';
+import { inject as service } from '@ember/service';
+import IntlService from 'ember-intl/services/intl';
+
+interface UploadAppStatusDetailsSignature {
+  Args: {
+    submission: SubmissionModel;
+  };
+}
+
+export default class UploadAppStatusDetailsComponent extends Component<UploadAppStatusDetailsSignature> {
+  @tracked iconName: string = 'downloading';
+  @tracked iconColor: string = 'download-icon-color';
+  @tracked icon: string = 'downloading';
+  @tracked timeDifference: string = '';
+  @service declare intl: IntlService;
+  intervalId: number;
+
+  constructor(owner: unknown, args: UploadAppStatusDetailsSignature['Args']) {
+
+    super(owner, args);
+    this.updateTimeDifference();
+    this.intervalId = setInterval(() => {
+        this.updateTimeDifference();
+    }, 60000);
+}
+
+
+  get runningStatus() {
+    const status = this.args.submission.status;
+    let obj = {
+      status: this.intl.t('completed'),
+      icon: 'download-done',
+      iconColor: 'success',
+      loading: true,
+      failed: false,
+    };
+    switch (status) {
+      case ENUMS.SUBMISSION_STATUS.DOWNLOAD_FAILED:
+      case ENUMS.SUBMISSION_STATUS.VALIDATE_FAILED:
+      case ENUMS.SUBMISSION_STATUS.STORE_URL_VALIDATION_FAILED:
+      case ENUMS.SUBMISSION_STATUS.STORE_DOWNLOAD_FAILED:
+      case ENUMS.SUBMISSION_STATUS.STORE_UPLOAD_FAILED:
+        obj.status = this.intl.t('failed');
+        obj.icon = 'error';
+        obj.iconColor = 'error';
+        obj.loading = false;
+        obj.failed = true;
+        return obj;
+      case ENUMS.SUBMISSION_STATUS.ANALYZING:
+        obj.status = this.intl.t('completed');
+        obj.icon = 'download-done';
+        obj.iconColor = 'success';
+        obj.loading = false;
+        obj.failed = false;
+        return obj;
+      default:
+        obj.status = this.intl.t('inProgress');
+        obj.icon = 'downloading';
+        obj.iconColor = 'download-icon-color';
+        obj.loading = true;
+        obj.failed = false;
+        return obj;
+    }
+  }
+
+  updateTimeDifference() {
+    this.timeDifference = dayjs(this.args.submission.createdOn).fromNow();
+  }
```

Comment:
if not a requirement to refresh then can be just refactored to `getter` and instead of `timeDifference` just call `createdOn`